mate-panel/panel-run-dialog.c

```diff
@@ -1945,6 +1993,26 @@ key_press_event (GtkWidget    *run_dialog,
 	return FALSE;
 }
 
+static void panel_run_dialog_accelerator_key_pressed(GtkAccelGroup *accel_group, GObject *acceleratable,
+                                               guint keyval, GdkModifierType modifier, PanelRunDialog *dialog)
+{
+	GtkTreeModel *model;
+        gint entry_number = -1;
+
+	model = GTK_TREE_MODEL (dialog->program_list_store);
+        for (gint i = 0; accelerator_key_mapping[i].list_item_idx != -1; i++) {
+                if (accelerator_key_mapping[i].key_id == keyval) {
+                        entry_number = accelerator_key_mapping[i].list_item_idx;
+                        break;
+                }
+        }
+        if (entry_number == -1) return;
+        GtkTreePath *path = gtk_tree_path_new_from_indices(entry_number, -1);
+        gtk_tree_view_set_cursor(GTK_TREE_VIEW(dialog->program_list), path, NULL, FALSE);
+        gtk_widget_grab_focus (dialog->program_list);
+        gtk_tree_path_free(path);
```

Comment:
I am ever so slightly worried that the match is not done on the accelerator actually displayed, although indeed that should match.  Maybe that's good enough though and looking up the rows would be a bit more annoying, not sure what makes more sense.

Reply:
Used a hashmap to obtain the proper entry. Unless I'm mistaken that should also reduce the computational complexity.